src/cli/mod.rs

```diff
@@ -361,95 +399,169 @@ impl FlakeHubPushCli {
                     .await
-                    .wrap_err("Getting token from JWT issuer's response")?;
-                response_deserialized.token
+                    .wrap_err("Getting upload bearer token from GitHub")?,
+
+                Some(jwt_issuer_uri) => {
+                    let client = build_http_client().build()?;
+                    let mut claims = github_actions_oidc_claims::Claims::make_dummy();
+                    // FIXME: we should probably fill in more of these claims.
+                    claims.aud = "flakehub-localhost".to_string();
+                    claims.iss = "flakehub-push-dev".to_string();
+                    claims.repository = repository.clone();
+                    claims.repository_owner = project_owner.to_string();
+                    claims.repository_id = github_graphql_data_result.project_id.to_string();
+                    claims.repository_owner_id = github_graphql_data_result.owner_id.to_string();
+
+                    let response = client
+                        .post(jwt_issuer_uri)
+                        .header("Content-Type", "application/json")
+                        .json(&claims)
+                        .send()
+                        .await
+                        .wrap_err("Sending request to JWT issuer")?;
+                    #[derive(serde::Deserialize)]
+                    struct Response {
+                        token: String,
+                    }
+                    let response_deserialized: Response = response
+                        .json()
+                        .await
+                        .wrap_err("Getting token from JWT issuer's response")?;
+                    response_deserialized.token
+                }
             }
+        } else {
+            return Err(eyre!(
+                "`flakehub-push` currently only runs inside Github Actions"
+            ));
+        };
+
+        // Here we merge explicitly user-supplied labels and the labels ("topics")
+        // associated with the repo. Duplicates are excluded and all
+        // are converted to lower case.
+        let labels: Vec<String> = labels
+            .into_iter()
+            .take(MAX_NUM_TOTAL_LABELS)
+            .map(|s| s.trim().to_lowercase())
+            .filter(|t: &String| {
+                !t.is_empty()
+                    && t.len() <= MAX_LABEL_LENGTH
+                    && t.chars().all(|c| c.is_alphanumeric() || c == '-')
+            })
+            .collect();
+
+        let Some(commit_count) = commit_count else {
+            return Err(eyre!("Could not determine commit count, this is normally determined via the `--git-root` argument or via the GitHub API"));
         };
 
         push_new_release(
             &host,
             &upload_bearer_token,
             &git_root,
             &subdir,
-            revision_info,
-            &repository,
+            revision,
+            commit_count,
```

Comment:
I'm going to keep looking to see if I can answer this, but it seems like maybe we use local commit count, but most times that's over-ridden by getting it from github?

I'm just a bit nervous this exposes us to (running outside github) and then having rolling releases be broken there.

(I'm looking to see if I've missed/forgotten a more fundamental fix to the local commit count)

Reply:
Any time a github token is present we rely on Github (since it is likely more correct -- and if we're in GHA then it's always 1).

If the user is running the tool outside GHA then we can't necessarily rely on being able to ask Github -- so we need to rely on the locally discovered commit count (which may be wrong).

Reply:
` (which may be wrong).`

totally understood, I just feel like we sorta still have an underlying bug that is mostly masked because we only expect to work in github. Yet we have an explicit code path that we know might return bad data.

It almost feels like we should be erroring out without a commit count we return. It feels like it's a `Maybe<T>` almost, unless we're in GitHub.

again, not an issue with this PR but I don't want us to merge this, forget, and then re-expose ourselves to this bug in a new codepath.